File: examples/lua53/client/protocols/AckSceneExit.lua
local M = {}
local moduleName = ...
_G[moduleName] = M
setmetatable(M, {__index = _G})
local _ENV = M


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckSceneExit})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uid = 0
end


function decode(self, pack)
	self.uid = pack:readU32()
	return self
end


function getUid(self)
	return self.uid
end

File: examples/javascript/client/proto/AckSceneExit.js
module.exports = AckSceneExit;

var Packet = require('../net/Packet');


function AckSceneExit() {
	this._uid = undefined;


	this.Decode = function(packet) {
		this._uid = packet.ReadUint();
	}


	this.SetUid = function(uid) {
		this._uid = uid;
	}
	this.GetUid= function() {
		return this._uid;
	}
}
